src/serlio/utils/MELScriptBuilder.h

```diff
@@ -73,10 +73,15 @@ class MELScriptBuilder {
 
 	void setsCreate(const MELVariable& setName);
 	void setsAddFaceRange(const std::wstring& setName, const std::wstring& meshName, int faceStart, int faceEnd);
+	void setsUseInitialShadingGroup(const std::wstring& meshName);
 
 	void createShader(const std::wstring& shaderType, const MELVariable& nodeName);
 	void createTextureShadingNode(const MELVariable& nodeName);
 
+	void cacheUndoState(const MELVariable& undoName);
+	void applyCachedUndoState(const MELVariable& undoName);
+	void setUndoState(bool undoState);

```

Comment:
the name "cache" is in this context a bit misleading - consider just calling it get/setUndoState

also, applyCachedUndoState and setUndoState look redundant

Reply:
We need a method for setting and getting the undo state. (first save the current setting, force undo state to off and later restore the previously defined setting)
But I agree, 2 functions get/setUndoState would suffice

Reply:
- updated names to get/setUndoState
- using overloads for setting from bool/mel variable